src/repo-query/pages/chat.tsx

```diff
@@ -0,0 +1,242 @@
+import { useEffect, useState } from "react";
+import { RepoQueryPages } from "../components/Dialog";
+import { REPO_QUERY_QUERY_ENDPOINT } from "../../constants";
+import ReactMarkdown from "react-markdown";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+import { darcula } from "react-syntax-highlighter/dist/esm/styles/prism";
+
+interface ChatMessage {
+    message: string;
+    isUser: boolean;
+}
+
+export const Chat = ({ ownerName, repoName, setCurrentPage }: { ownerName: string, repoName: string, setCurrentPage: (page: RepoQueryPages) => void }) => {
+    // chat ui with bubbles and input
+    const [chatHistory, setChatHistory] = useState<ChatMessage[]>([]);
+    const [sendEnabled, setSendEnabled] = useState(true);
+    const [botThinking, setBotThinking] = useState(false);
+    const [statusMessage, setStatusMessage] = useState("Thinking...🧠");
+
+    const processChunk = (chunk: string) => {
+        const chunkLines = chunk.split("\n");
+        const eventLine = chunkLines[0];
+        const dataLine = chunkLines[1];
+
+        const event = eventLine.split(": ")[1];
+        let data = { path: "" };
+
+        try {
+            data = JSON.parse(dataLine.split(": ")[1]);
+        } catch (e) {
+            // untermindated stirng
+            console.log(e, dataLine);
+        }
+
+
+        if (event === "SEARCH_CODEBASE") {
+            setStatusMessage("Searching the codebase for your query...🔍");
+        } else if (event === "SEARCH_FILE") {
+            setStatusMessage(`Searching ${data.path} for your query...🔍`);
+        } else if (event === "GENERATE_RESPONSE") {
+            setStatusMessage("Generating a response...🧠");
+        } else if (event === "DONE") {
+            setBotThinking(false);
+            setSendEnabled(true);
+            setChatHistory(prevChatHistory => [...prevChatHistory, {
+                message: data as unknown as string,
+                isUser: false,
+            }]);
+        }
+    };
+
+    const askQuery = async (query: string) => {
+        setBotThinking(true);
+        setSendEnabled(false);
+        const response = await fetch(
+            `${REPO_QUERY_QUERY_ENDPOINT}`,
+            {
+                method: "POST",
+                headers: { "Content-Type": "application/json" },
+                body: JSON.stringify({
+                    query,
+                    repository: {
+                        owner: ownerName,
+                        name: repoName,
+                        branch: "beta",
```

Comment:
```suggestion
                        branch: "HEAD"
```

Reply:
so HEAD will work for all repositories?

Reply:
This is how we're getting the repos.

https://github.com/open-sauced/repo-query/blob/0e9fcac61f7870ac69892f6d8128acd93683b89d/src/github/mod.rs#L88

So using HEAD for the branch name, https://github.com/open-sauced/ai/archive/HEAD.zip

Fetches the default branch. 👍👍

Reply:
We also have a utility function for it too.
https://github.com/open-sauced/ai/blob/beta/src/utils/urlMatchers.ts#L69-L74

Reply:
great.